rust/parser/src/lexer.rs

```diff
@@ -1381,4 +1381,12 @@ impl<'alloc> AutoCow<'alloc> {
             None => &self.start[..self.start.len() - lexer.chars.as_str().len()],
         }
     }
+
+    // Just like finish, but without pushing current char.
+    fn finish_without_push(&mut self, lexer: &Lexer<'alloc>) -> &'alloc str {
+        match self.value.take() {
+            Some(arena_string) => arena_string.into_bump_str(),
+            None => &self.start[..self.start.len() - lexer.chars.as_str().len() - 1],
```

Comment:
Hmm. `- 1` is right only because, at each call site, we know that the current char is < 128. I don't see an easy way to make this more foolproof though.